es6/README.md

```diff
@@ -48,9 +80,200 @@ const MaxAllowed = 7;
+
+For more on constants, read [_Learning ES6: Block-level scoping with `let` and `const`_](http://www.eventbrite.com/engineering/learning-es6-block-level-scoping-with-let-and-const/).
+
+**[⬆ back to top](#table-of-contents)**
+
+## Strings
+
+### Template literals
+
+When building up a string, use a template literal instead of string concatenation (eslint: [`prefer-template`](http://eslint.org/docs/rules/prefer-template.html)):
+
+```js
+// good
+const generateGreeting = (name=DEFAULT_NAME) => {
+    const formattedNow = (new Date()).toString();
+
+    return `Hi, ${name} on ${formattedNow}`;
+}
+
+// bad (uses string concatenation)
+const generateGreeting = (name=DEFAULT_NAME) => {
+    const formattedNow = (new Date()).toString();
+
+    return 'Hi, ' + name + ' on ' + formattedNow;
+}
+
+// bad (uses array join)
+const generateGreeting = (name=DEFAULT_NAME) => {
+    const formattedNow = (new Date()).toString();
+
+    return ['Hi, ', name, ' on ', formattedNow].join();
+}
+```
+
+When using template literals, tokens should **not** be padded by spaces (eslint: [`template-curly-spacing`](http://eslint.org/docs/rules/template-curly-spacing)):
+
+```js
+// good
+const generateGreeting = (name=DEFAULT_NAME) => {
+    const formattedNow = (new Date()).toString();
+
+    return `Hi, ${name} on ${formattedNow}`;
+}
+
+// bad (has extra padding around the curlies)
+const generateGreeting = (name=DEFAULT_NAME) => {
+    const formattedNow = (new Date()).toString();
+
+    return `Hi, ${ name } on ${ formattedNow }`;
+}
+```
+
+Use template literals instead of escape single quotes (eslint: [`no-useless-escape`](http://eslint.org/docs/rules/no-useless-escape)):
+
+```js
+// good
+let message = `He's the one!`;
+
+// bad (escapes the single quote)
+let message = 'He\'s the one!';
+
+// bad (uses double quotes to avoid quoting)
+let message = "He's the one!";
+```
+
+For more on template literals, read [_Learning ES6: Template literals & tagged templates_](http://www.eventbrite.com/engineering/learning-es6-template-literals-tagged-templates/).
+
+**[⬆ back to top](#table-of-contents)**
+
+## Arrays
+
+### Array + spread operator
+
+Use the spread operator (`...`) to copy an array:
+
+```js
+// good
+let list = [1, 2, 3];
+let listCopy = [...list];
+
+// bad (uses `concat` to copy)
+let list = [1, 2, 3];
+let listCopy = list.concat();
+
+// bad (uses a map to create a copy)
+let list = [1, 2, 3];
+let listCopy = list.map((item) => item);
+```
+
+Use the spread operator (`...`) to join multiple arrays together:
+
+```js
+// good
+let start = ['do', 're', 'mi'];
+let end = ['la', 'ti'];
+let scale = [...start, 'fa', 'so', ...end];
+
+// bad
+let start = ['do', 're', 'mi'];
+let end = ['la', 'ti'];
+let scale = start.concat(['fa', 'so']).concat(end);
+```
+
+Use the spread operator (`...`) to convert an array-like object into an `Array`:
+
+```js
+// NodeList object
+let nodeList = document.querySelectorAll('p');
+
+// Array
+let nodes = [...nodeList];
```

Comment:
we are missing the bad here, we might wanna just do a for loop to expose the bad idea.